_posts/2025-05-08-opensearch-performance-3.0.md

```diff
@@ -0,0 +1,1013 @@
+
+* **[Support for BFloat16 (efficient fp16) for vectors](https://github.com/opensearch-project/k-NN/issues/2510)**: The fp16 support using Faiss scalar quantizer(SQfp16) reduces memory usage by 50% while maintaining performance and recall levels similar to fp32 vectors. However, it has a limitation: input vectors must fall within the range of [-65504, 65504]. BFloat16 offers an alternative that provides the full range of fp32 by trading off precision (it supports up to 2 or 3 decimal values, or 7 mantissa bits) and still uses 16 bits per dimension (providing a 50% memory reduction).
+
+* **[Multi-tenancy/high-cardinality filtering vector search](https://github.com/opensearch-project/k-NN/issues/1140)**: OpenSearch is taking multi-tenant vector search to the next level with its enhanced filtering capabilities. At the core of this improvement is an innovative approach to partitioning HNSW graphs based on tenant boundaries. This strategic enhancement delivers two critical advantages: superior search accuracy and robust data isolation between tenants. By organizing vector search data structures according to tenant boundaries, filter queries operate exclusively within their respective tenant partitions, eliminating cross-tenant noise and improving precision.
+
+* **Improving vector search throughput with smart query routing**: By implementing semantic-based data organization across OpenSearch nodes, grouping similar embeddings together, we can enable targeted searching of relevant nodes for nearest neighbors. This approach allows searching a subset of shards per query, potentially increasing throughput by up to 3x while reducing compute requirements.
+
+OpenSearch 3.1 will be the first minor release to build on 3.0 based on the [OpenSearch release schedule](https://opensearch.org/releases/). It will include Lucene updates, refinements to concurrent search, and improvements to aggregation, vector and indexing frameworks. As usual, we'll share performance updates in the release notes and blogs.
+
+
+## Appendix: Benchmarking methodology
+
+All performance comparisons were conducted using a repeatable process based on the OpenSearch Benchmark (OSB) tool and the Big 5 workload. Benchmarks covered match queries, term aggregations, range filters, date histograms, and sorted queries. The dataset (~100 GB, 116 million documents) reflects time-series and e-commerce use cases.
+
+**Environment**: Tests were run on a single-node OpenSearch cluster using c5.2xlarge EC2 instances (8 vCPUs, 16 GB RAM, 8 GB JVM heap). Default settings were used unless noted. Indexes had one primary shard and no replicas to avoid multi-shard variability. Documents were ingested chronologically to simulate time-series workloads.
```

Comment:
```suggestion
**Environment**: Tests were run on a single-node OpenSearch cluster using c5.2xlarge Amazon Elastic Compute Cloud (Amazon EC2) instances (8 vCPUs, 16 GB RAM, 8 GB JVM heap). Default settings were used unless noted. Indexes had one primary shard and no replicas to avoid multi-shard variability. Documents were ingested chronologically to simulate time-series workloads.
```